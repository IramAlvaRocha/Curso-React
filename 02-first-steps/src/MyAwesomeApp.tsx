import type { CSSProperties } from "react";

const firstName = 'Fernando';
const lastName = 'Herrera';

const favoriteGames = ["Elden Ring", "Hades", "Celeste"];
const isActive = true;

const address = {
    zipCode: 'ABC-123',
    city: "CanadÃ¡",
}

const myStyles: CSSProperties={
    backgroundColor: "#fafafa",
    borderRadius: 20,
    padding: 10,
    marginTop: 30
}

export const MyAwesomeApp  = () => {
  return (
    <div data-testid="div-app">
      <h1 data-testid="first-name-test"> {firstName} </h1>
      <h3> {lastName} </h3>

      <p>{2+2}</p>
      <p className="mi-clase-favorita">{favoriteGames.join(", ")}</p>

      <h1>{isActive ? 'Activo' : 'No activo'}</h1>

      <p style={myStyles}>{JSON.stringify(address)}</p>
    </div>
  )
}
