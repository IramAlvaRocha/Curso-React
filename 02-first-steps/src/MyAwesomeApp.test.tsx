import { describe, expect, test } from "vitest";
import {render, screen} from "@testing-library/react"
import { MyAwesomeApp } from "./MyAwesomeApp";

describe("MyAwesomeApp", ()=>{
    test('Should render first name and last name', ()=>{       
        const { container } = render(<MyAwesomeApp />);
        // console.log(container.innerHTML);
        // screen.debug();

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');

        //ESTAS SON LAS PRUEBAS UTILIZANDO CONTAINER QUE DESESTRUCTURAMOS DEL RENDER()
        //Que contenga en este caso se ignoran los espacios que estan en el componente, para buscar solamente la palabra Fernando 
        //y la prueba no falle porque hacen falta los espacios del elemento
        expect(h1?.innerHTML).toContain('Fernando');
        expect(h3?.innerHTML).toContain('Herrera');

    })
    
    test('Should render first name and last name - SCREEN', ()=>{       
        render(<MyAwesomeApp />);
        // console.log(container.innerHTML);
        screen.debug();

        // const h1 = screen.getByRole('heading', {
        //     level: 1
        // });

        const h1 = screen.getByTestId("first-name-test");

        expect(h1.innerHTML).toContain("Fernando")
    })


    test('should match snapshot', ()=>{
        const {container} = render(<MyAwesomeApp />);

        expect(container).toMatchSnapshot()
    })
   
    test('should match snapshot', ()=>{
         render(<MyAwesomeApp />);
         expect(screen.getByTestId("div-app")).toMatchSnapshot()
    })

})
