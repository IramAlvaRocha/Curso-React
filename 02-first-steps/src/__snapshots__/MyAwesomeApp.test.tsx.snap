// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-test"
    >
       
      Fernando
       
    </h1>
    <h3>
       
      Herrera
       
    </h3>
    <p>
      4
    </p>
    <p
      class="mi-clase-favorita"
    >
      Elden Ring, Hades, Celeste
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {"zipCode":"ABC-123","city":"Canadá"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-test"
  >
     
    Fernando
     
  </h1>
  <h3>
     
    Herrera
     
  </h3>
  <p>
    4
  </p>
  <p
    class="mi-clase-favorita"
  >
    Elden Ring, Hades, Celeste
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {"zipCode":"ABC-123","city":"Canadá"}
  </p>
</div>
`;
